### PR TITLE
Magiclysm: Alchemy, Part 1-Potion tools and Stat Potion Brewing

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Mods "Magiclysm: First additions to alchemy"```

#### Purpose of change
Potions spawning in the world imply there was a creation process to them.

#### Describe the solution
Adds two new tools utilizing the new tool qualities from #32051 for alchemy, a book with recipes in it for one tool and the four stat potions, a new harvest from large monsters and small furred mammals, and adjusts itemgroups to spawn in the new book. Additionally, the rune spells from #31687 are enabled to be learned upon taking a school.

#### Describe alternatives you've considered
Assuming the potions appeared fully made from nowhere and were collected in shops.

### CODE REVIEW

File: data/mods/Magiclysm/recipes/recipe_potions.json [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    "difficulty": 3,
+    "charges": 7,
+    "time": "120 m",
+    "batch_time_factors": [ 75, 4 ],
+    "book_learn": [ [ "alchemy_basic", 3 ] ],
+    "qualities": [ { "id": "CHEM", "level": 1 }, { "id": "CONCENTRATE", "level": 1 }, { "id": "MANA_INFUSE", "level": 1 } ],
+    "tools": [ [ [ "surface_heat", 10, "LIST" ] ] ],
+    "components": [
+      [
+        [ "gin", 7 ],
```
Comment (KorGgenT) [inline data/mods/Magiclysm/recipes/recipe_potions.json]: there should be a requirement for this, i think you should use that instead.

File: data/mods/Magiclysm/recipes/recipe_potions.json [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    "difficulty": 5,
+    "time": "60 m",
+    "batch_time_factors": [ 50, 4 ],
+    "book_learn": [ [ "alchemy_basic", 4 ] ],
+    "qualities": [ { "id": "CHEM", "level": 2 }, { "id": "CONCENTRATE", "level": 1 }, { "id": "MANA_FOCUS", "level": 1 } ],
+    "tools": [ [ [ "surface_heat", 10, "LIST" ] ], [ [ "rune_alchemist", -1 ], [ "rune_magus", -1 ] ] ],
+    "components": [
+      [ [ "potion_starter", 5 ] ],
+      [
+        [ "egg_bird", 5 ],
```
Comment (KorGgenT) [inline data/mods/Magiclysm/recipes/recipe_potions.json]: iirc there should be a requirement for eggs. i think you should use that.

File: data/mods/Magiclysm/recipes/recipe_potions.json (hunk 1 above)
Comment (LaVeyanFiend) [reply data/mods/Magiclysm/recipes/recipe_potions.json]: Eggs_bird doesn't respect mass of each egg being differrent, it counts each egg as 1 regardless of size.

File: data/mods/Magiclysm/recipes/recipe_potions.json (hunk 2 above)
Comment (LaVeyanFiend) [reply data/mods/Magiclysm/recipes/recipe_potions.json]: Eggs_bird doesn't respect how large each egg actually is, when the larger eggs are 83ml and the smaller ones are 5ml.

File: data/mods/Magiclysm/recipes/recipe_potions.json (hunk 1 above)
Comment (LaVeyanFiend) [reply data/mods/Magiclysm/recipes/recipe_potions.json]: Changed to hard_liquor.

File: data/mods/Magiclysm/items/comestibles.json [hunk 3]
```diff
@@ -32,5 +32,24 @@
...
+  },
+  {
+    "id": "hairball",
+    "type": "COMESTIBLE",
+    "comestible_type": "MED",
+    "healthy": -1,
+    "stack_size": 1,
+    "material": [ "fur", "water" ],
+    "primary_material": "fur",
+    "volume": "5ml",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/items/comestibles.json]:
```suggestion
    "volume": "350ml",
```

File: data/mods/Magiclysm/items/comestibles.json [hunk 4]
```diff
@@ -32,5 +32,24 @@
...
+    "stack_size": 1,
+    "material": [ "fur", "water" ],
+    "primary_material": "fur",
+    "volume": "350ml",
+    "weight": 15,
+    "color": "brown",
+    "symbol": "0",
+    "name": "hairball",
+    "name_plural": "hairballs",
+    "description": "A big ugly ball of animal spit and licked-off hairs. Don't swallow it.",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/items/comestibles.json]:
```suggestion
    "description": "A big, ugly ball of animal spit and licked-off hairs. Don't swallow it.",
```

**Pull request merged**